lib/nrf_modem_lib/shell/CMakeLists.txt

```diff
@@ -0,0 +1,7 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+add_subdirectory_ifdef(CONFIG_NRF_MODEM_LIB_SHELL_TRACE trace)
```

Comment:
@lemrey Regarding your [comment](https://github.com/nrfconnect/sdk-nrf/pull/13725#discussion_r1476078782) (I'm not allowed to respond on an outdated file): Is this an OK alternative? I've moved the trace related code to a subdirectory, so now it's only one `ifdef` for trace in the shell folder´s `CMakeLists.txt`. But I've kept the UART logic in a separate C file to avoid having severa `#ifdef`s around the UART logic in trace.c.

Reply:
I'd like to keep it simple for now and try to have one C file for these shell modules.